File: examples/lh-server/python/03-variables/workflow.py
from littlehorse.workflow import Workflow, WorkflowThread


def get_workflow() -> Workflow:
    def wfLogic(wf: WorkflowThread) -> None:
        user_id = wf.declare_str("user-id").required().searchable()
        user = wf.declare_json_obj("user")
        age = wf.declare_int("age")
        fetched_user = wf.execute("fetch-user", user_id)
        user.assign(fetched_user)
        age.assign(fetched_user.with_json_path("$.age"))
        message = wf.format(
            "Hello there, {0}! You are {1} years old",
            user.with_json_path("$.title"),
            age,
        )
        wf.execute("send-email", user.with_json_path("$.email"), message)

    return Workflow("variables-example", wfLogic)

File: examples/lh-server/go/03-variables/workflow.go
package main

import "github.com/littlehorse-enterprises/littlehorse/sdk-go/littlehorse"

func Wflogic(wf *littlehorse.WorkflowThread) {
	userId := wf.DeclareStr("user-id").Required().Searchable()
	userObject := wf.DeclareJsonObj("user-obj")
	age := wf.DeclareInt("age")
	userOutput := wf.Execute("fetch-user", userId)

	userObject.Assign(userOutput)
	age.Assign(userOutput.JsonPath("$.age"))
	message := wf.Format(
		"Hello there, {0}! You are {1} years old",
		userId, age,
	)
	wf.Execute("send-email", userObject.JsonPath("$.email"), message)
}
